addons/nuxeo-core-binarymanager-cloud/nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/bulk/S3SetContentLengthAction.java

```diff
@@ -0,0 +1,178 @@
+import org.apache.logging.log4j.Logger;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.PropertyException;
+import org.nuxeo.ecm.core.api.VersioningOption;
+import org.nuxeo.ecm.core.api.impl.DownloadBlobGuard;
+import org.nuxeo.ecm.core.api.versioning.VersioningService;
+import org.nuxeo.ecm.core.blob.BlobInfo;
+import org.nuxeo.ecm.core.blob.BlobManager;
+import org.nuxeo.ecm.core.blob.BlobProvider;
+import org.nuxeo.ecm.core.blob.ManagedBlob;
+import org.nuxeo.ecm.core.blob.SimpleManagedBlob;
+import org.nuxeo.ecm.core.bulk.action.computation.AbstractBulkComputation;
+import org.nuxeo.ecm.core.storage.sql.S3BinaryManager;
+import org.nuxeo.lib.stream.computation.Topology;
+import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.stream.StreamProcessorTopology;
+
+/**
+ * Bulk Action to update the file:content/length property using the value in the binary store.
+ *
+ * @since 2021.8
+ */
+public class S3SetContentLengthAction implements StreamProcessorTopology {
+
+    public static final String ACTION_NAME = "s3SetContentLength";
+
+    public static final String DISABLE_QUOTA_CHECK_LISTENER = "disableQuotaListener";
+
+    public static final String DISABLE_DUBLINCORE_LISTENER = "disableDublinCoreListener";
+
+    public static final String DISABLE_NOTIFICATION_SERVICE = "disableNotificationService";
+
+    public static final String DISABLE_THUMBNAIL_SERVICE = "disableThumbnailComputation";
+
+    public static final String DISABLE_AUTO_CHECKOUT = "DisableAutoCheckOut";
+
+    public static final String DISABLE_DOMAIN_LISTENER = "disableDomainListener";
+
+    public static final String DISABLE_HTMLSANITIZER_LISTENER = "disableHtmlSanitizerListener";
+
+    public static final String DISABLE_AUTO_INDEXING = "disableAutoIndexing";
+
+    private static final String FORCE_OPTION = "force";
+
+    @Override public Topology getTopology(Map<String, String> options) {
+        return Topology.builder()
+                       .addComputation(SetContentLengthComputation::new,
+                               Arrays.asList(INPUT_1 + ":" + ACTION_NAME, OUTPUT_1 + ":" + STATUS_STREAM))
+                       .build();
+    }
+
+    public static class SetContentLengthComputation extends AbstractBulkComputation {
+
+        private static final Logger log = LogManager.getLogger(SetContentLengthComputation.class);
+
+        public SetContentLengthComputation() {
+            super(ACTION_NAME);
+        }
+
+        protected String currentProvider;
+
+        @Override protected void compute(CoreSession session, List<String> ids, Map<String, Serializable> properties) {
+            DownloadBlobGuard.enable(); // Avoid downloading main content blob during the transaction
+            boolean force = properties.containsKey(FORCE_OPTION) && Boolean.parseBoolean(
+                    properties.get(FORCE_OPTION).toString());
```

Comment:
```suggestion
            boolean force = properties.containsKey(FORCE_OPTION) && (Boolean)properties.get(FORCE_OPTION);
```

Reply:
properties come from command input the above will not work si^prt {"force":"true"} ?